test/parallel/test-http2-compat-serverresponse-statusmessage.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+// Http2ServerResponse.writeHead should accept an optional status message
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  const port = server.address().port;
+  server.once('request', common.mustCall(function(request, response) {
+    const statusCode = 200;
+    const statusMessage = 'OK';
+    const headers = {'foo-bar': 'abc123'};
+    common.expectWarning(
```

Comment:
If it's avoiding it it's because of a race condition that you're fortunate enough to be winning ;-) ... all `common.expectWarning()` does is add a listener to `process.on('warning')`. I'm not sure if it's currently set up to support listening for two types of warnings so it may be necessary to use your own `process.on('warning') handler instead (that can listen to either)